core/crypto/src/signature.rs

```diff
@@ -508,13 +506,13 @@ impl Signature {
         signature_data: &[u8],
     ) -> Result<Self, crate::errors::ParseSignatureError> {
         match signature_type {
-            KeyType::ED25519 => Ok(Signature::ED25519(
-                ed25519_dalek::Signature::from_bytes(signature_data).map_err(|err| {
+            KeyType::ED25519 => Ok(Signature::ED25519(ed25519_dalek::Signature::from_bytes(
+                <&[u8; 64]>::try_from(signature_data).map_err(|err| {
```

Comment:
Either elide the `64` or use a relevant constant? Hm. Having 64 here is pretty nice too as it allows to detect breaking changes. Not sure which is better...

Reply:
Just to say: AFAICT this is only ever used in rosetta-rpc, and so is not important for protocol version (if not this would deserve the binary check too)